### PR TITLE
FPO-158: Configure Sentry and structured logs

### PR DESCRIPTION
**Pull request by willfish**
### Jira link

FPO-158

### What?

I have added/removed/altered:

- Adds sentry package
- Adds configuration for sentry
- Adds configuration for morgan middleware logging
- Adds configuration for simple error handling
- Separate development/production dependencies to reduce our surface area for vulnerabilities
- Fix broken linting behaviour

### Why?

I am doing this because:

- This is required as part of productionising this application

### AC

- Tested
- Shared with stakeholders
- Pair reviewed

### CODE REVIEW

File: src/app.ts [hunk 1]
```diff
@@ -1,72 +1,111 @@
...
-if (isDev) {
-  const morgan = require('morgan')
-  app.use(morgan('dev'))
+const sentryDsn = process.env.SENTRY_DSN ?? ''
+const sentryEnv = process.env.SENTRY_ENVIRONMENT ?? ''
+
+async function loadDev (): Promise<void> {
+  if (isDev) {
+    // eslint-disable-next-line @typescript-eslint/no-var-requires
+    const swaggerJsdoc = require('swagger-jsdoc')
```
Comment (willfish) [inline src/app.ts:22]: To dynamically load dependencies I had to put these requires behind an async function

Comment (willfish) [reply src/app.ts:22]: I don't want them in the production image since these are for local development, only

Comment (shjohnson) [reply src/app.ts:22]: yeah nice, Sounds good!

Comment (shjohnson) [reply src/app.ts:22]: guessing still works without the imports?

Comment (willfish) [reply src/app.ts:22]: It does, yeah. Tested locally :tada:

**Review by shjohnson — APPROVED** [review]
LGTM 🍏

**Pull request merged**